xbmc/platform/linux/AppParamParserLinux.cpp

```diff
@@ -82,12 +90,29 @@ void CAppParamParserLinux::ParseArg(const std::string& arg)
       exit(0);
     }
   }
+  else if (arg.substr(0, 16) == "--audio-backend=")
+  {
+    const auto it =
+        std::find(availableAudioBackends.cbegin(), availableAudioBackends.cend(), arg.substr(16));
+    if (it != availableAudioBackends.cend())
+    {
+      GetAppParams()->SetAudioBackend(*it);
+    }
+    else
+    {
+      std::cout << StringUtils::Format(audioBackendsText, arg.substr(15),
+                                       StringUtils::Join(availableAudioBackends, ", "));
+
+      exit(0);
+    }
+  }
```

Comment:
This also fixes the `arg.substr(15)`:
```suggestion
  else if (arg.substr(0, 16) == "--audio-backend=")
  {
    const auto argValue = arg.substr(16);
    if (std::any_of(availableAudioBackends.cbegin(), availableAudioBackends.cend(),
                    [&](const auto& v) { return v == argValue; }))
    {
      GetAppParams()->SetAudioBackend(argValue);
    }
    else
    {
      std::cout << StringUtils::Format(audioBackendsText, argValue,
                                       StringUtils::Join(availableAudioBackends, ", "));

      exit(0);
    }
  }
```

Reply:
partially updated.